Remove mention of deprecated I/O methods from Predef main comment

janekdb opened the pull request:
The read\* functions are deprecated so should not be highlighted.

soc commented:
I'd probably drop printf too, considering we have the `f` interpolator.

janekdb commented:
`printf` removed from main comment.

If the aim is to move attention to the `f` interpolator` then deprecation of `Predef.printf` needs to be considered (in a future commit).

Is this too strong? I wonder how much use `printf` gets. Enough to justify remaning in `Predef`?

som-snytt commented:
I never use `Predef.printf`, even though I call what I do `printf`, out of habit.

Sorry to bikeshed, or sorry to raze the shed, but the whole paragraph is now wrongish. 

```
For basic console output, `Predef` provides convenience methods `print` and `println`,
which are aliases of the methods in the object `scala.Console`.
```
If anybody knows how to turn `print` into a link, it should be one. Brackets? Bracken?

Probably the section title should be `Console Output` or something innocuous.

In a separate PR, really the whole doc could be rewritten, to point out `any2stringadd` and how to disable, `???`, etc. Also mention the compiler option `-Yno-predef`, maybe at the end in a footnote or implementation note.

janekdb commented:
I'll linkify `print` and `println` shorty but in the meanwhile could you be more specific about how the paragraph is wrongish? Even provide a full rewrite of the paragraph it that's easier.

som-snytt commented:
My comment did provide a rewrite in triple-backticks.

janekdb commented:
@som-snytt - Used your text. Decided to copy over the method Scaladoc from `Console` instead of making the method mentions links. I think this is more likely to benefit a user in a hurry.

This is how it looks,

![scaladoc-2 12-predef-no-read-methods-main-comment](https://cloud.githubusercontent.com/assets/1123855/13651841/0b5c4e72-e641-11e5-8d02-897d576d60a4.png)

![scaladoc-2 12-predef-no-read-methods-methods](https://cloud.githubusercontent.com/assets/1123855/13651852/1d689b5c-e641-11e5-9ff1-0fbe61896104.png)

som-snytt commented:
I actually meant links to `Predef.println` from the text, which is even more useful now that you have added doc there. :)  I'm thinking of the new user who doesn't know to tab to the search box etc.  I suggest that link only if you really want to see your face reflected by the high polish of the PR; I don't mean to nit-pick; and the other methods referenced up there don't have links. It looks great.

janekdb commented:
@som-snytt The note about `f` being idiomatic has been added. Over to reviewers now.

som-snytt commented on src/library/scala/Predef.scala:
```diff
@@ -304,9 +302,38 @@ object Predef extends LowPriorityImplicits with DeprecatedPredef {
+  /** Prints an object to `out` using its `toString` method.
+    *
+    *  @param x the object to print; may be null.
+    */
```
The asterisk should align with the first of the double stars. See the top comment. Javadoc style (see second comment in source) seems to sneak by, but alt-Scaladoc style is out of favor. I'm not the comment cop, but since this is `Predef`, uniformity is worth enforcing. Is there a SO Q&A, "How many ways are there to write Scaladoc comments in Scala?" Luckily, April 1 is just around the corner.

janekdb commented:
Alignment problems fixed. IDEA being overly helpful and reformatting copy/paste. Grrrr!

With that blemish eradicated the time has come for fresh dusting of fingerprint power. Have we yet commissioned the perfect crime?

som-snytt commented:
LGTM

I might have put guidance ("Consider using something else") after the method description; but that's a matter of emphasis.

janekdb commented:
Moved the guidance to after the method description.

som-snytt commented:
You are the Terminator! "He never gets tired, he'll never stop coming after you!" Actually, the quote I'm thinking of is T2, "It would never stop, it would never leave him, it would never say it was too busy to make one more doc change in a PR."  I amended that last part. Well, Sarah Connor goes on to say, "In an insane world, it was the sanest choice."

adriaanm commented:
/synch

SethTisue commented on src/library/scala/Predef.scala:
```diff
@@ -304,9 +302,37 @@ object Predef extends LowPriorityImplicits with DeprecatedPredef {
+   *  <a href="" target="contentFrame" class="java/util/Formatter">
+   *  `java.util.Formatter`</a>.
+   *
+   *  Consider using the [[scala.StringContext.f f interpolator]] as more idiomatic.
```
suggest: "more typesafe and idiomatic"

som-snytt replied:
Now more Typomatic!

Is it too late to change from lightbend to typomatic?

janekdb committed:
Remove mention of deprecated I/O methods from Predef main comment

The read* functions are deprecated so should not be highlighted.

Copied Scaladoc from Console on the assumption that few people will tarry
for long once they have seen the existence of the print* methods.

Linkified print* methods.

Added note to printf encouraging use of f interpolator.

janekdb commented:
Made the "type safe and" improvement. Thanks!

SethTisue commented:
thank you, Janek!

SethTisue commented:
note that discussion on the possible deprecation of `Predef.println` is at https://groups.google.com/forum/#!topic/scala-user/1NLGIvVbQdg
